hls4ml/backends/vivado/passes/pointwise.py

```diff
@@ -4,13 +4,45 @@
     Conv1DFunctionTemplate,
     Conv2DConfigTemplate,
     Conv2DFunctionTemplate,
-    conv1d_config_template,
     conv2d_config_template,
     conv_mult_config_template,
 )
 from hls4ml.model.layers import register_layer
 from hls4ml.model.optimizer import OptimizerPass
 
+pointwise_conv1d_config_template = """struct config{index} : nnet::conv1d_config {{
+    static const unsigned pad_left = {pad_left};
+    static const unsigned pad_right = {pad_right};
+    static const unsigned in_width = {in_width};
+    static const unsigned n_chan = {n_chan};
+    static const unsigned filt_width = {filt_width};
+    static const unsigned kernel_size = filt_width;
+    static const unsigned n_filt = {n_filt};
+    static const unsigned stride_width = {stride_width};
+    static const unsigned dilation = {dilation};
+    static const unsigned out_width = {out_width};
+    static const unsigned reuse_factor = {reuse};
+    static const unsigned n_zeros = {nzeros};
+    static const bool store_weights_in_bram = false;
+    static const unsigned strategy = nnet::{strategy};
+    static const nnet::conv_implementation implementation = nnet::conv_implementation::{implementation};
+    static const unsigned min_width = {min_width};
+    static const ap_uint<filt_width> pixels[min_width];
+    static const unsigned n_partitions = {n_partitions};
+    static const unsigned n_pixels = out_width / n_partitions;
+    template<class data_T, class CONFIG_T>
+    using fill_buffer = nnet::{fill_fn}<data_T, CONFIG_T>;
+    typedef {accum_t.name} accum_t;
+    typedef {bias_t.name} bias_t;
+    typedef {weight_t.name} weight_t;
+    typedef {config_t} mult_config;
+    template<unsigned K, unsigned S, unsigned W>
+    using scale_index = nnet::{scale_index_type}<K, S, W>;
+    template<class data_T, class res_T, class CONFIG_T>
+    using pointwise_conv = nnet::{pointwise_fn}<data_T, res_T, CONFIG_T>;
```

Comment:
In `Dense` layers we moved to using a function pointer like this that the main `dense()` function calls, eliminating the need for checks in HLS. Here it would also simplify the call hierarchy (no need for special handling of pointwise) and no need for this template.

Reply:
attempted in https://github.com/jmduarte/hls4ml/pull/22/commits/d56dc7349e94ad8b6aaefe5b0f22385d3d9a9e03